addons/account/views/report_invoice.xml

```diff
@@ -181,13 +181,10 @@
                     <p t-if="o.invoice_incoterm_id" name="note">
                         <strong>Incoterm: </strong><span t-field="o.invoice_incoterm_id.code"/> - <span t-field="o.invoice_incoterm_id.name"/>
                     </p>
-                    <div id="qrcode" t-if="(o.company_id.qr_code) and (o.currency_id.name == 'EUR') and (o.invoice_partner_bank_id.acc_number != False)">
-                        <p t-if="(o.invoice_partner_bank_id.qr_code_valid)">
-                            <strong class="text-center">Scan me with your banking app.</strong><br /><br />
-                            <img class="border border-dark rounded" t-att-src="o.invoice_partner_bank_id.build_qr_code_url(o.amount_residual,(o.ref) if (o.ref) else o.name)"/>
-                        </p>
-                        <p t-if="(o.invoice_partner_bank_id.qr_code_valid == False)">
-                            <strong class="text-center">The SEPA QR Code informations are not set correctly.</strong><br />
+                    <div id="qrcode" t-if="o.display_qr_code">
+                        <p t-if="qr_code_urls.get(o.id)">
+                            <strong class="text-center">Scan me with your banking app.</strong><br/><br/>
+                            <img class="border border-dark rounded" t-att-src="qr_code_urls[o.id]"/>
```

Comment:
why don't you call directly `o.generate_qr_code()` in the xml?

Reply:
Because it can raise and UserError in case something is misconfigured, and if it raises during qweb rendering, you get a whole traceback (with the error, but still) instead of a clean error message. This isn't acceptable for users.

Reply:
I see the kind of traceback you are talking about... :disappointed: